pkg/controller/elasticsearch/validation/validation.go

```diff
@@ -13,6 +13,7 @@ import (
 
 const (
 	cfgInvalidMsg            = "configuration invalid"
+	validationFailedMsg      = "Spec validation failed"
```

Comment:
This needs to be more specific - any suggestions? Also, should it be general or specific to the issue we've found (as we already know it)?

Reply:
I would make it specific, like:
`Spec validation failed: spec.nodeSets undefined`
`Spec validation failed: spec.nodeSets.count undefined`

(undefined, not found or invalid?)

Reply:
I'll go with invalid here since we now don't know if it's not specified or empty.